ui/pages/confirmations/components/confirm/info/hooks/use-selected-token.ts

```diff
@@ -0,0 +1,25 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useSelector } from 'react-redux';
+import { toChecksumHexAddress } from '../../../../../../../shared/modules/hexstring-utils';
+import {
+  getAllTokens,
+  getCurrentChainId,
+  getSelectedAccount,
+} from '../../../../../../selectors';
+
+export const useSelectedToken = (transactionMeta: TransactionMeta) => {
+  const selectedAccount = useSelector(getSelectedAccount);
+
+  const detectedTokens = useSelector(getAllTokens);
+  const chainId = useSelector(getCurrentChainId);
+
+  const selectedToken = detectedTokens?.[chainId]?.[
```

Comment:
Minor, but should be this be in a selector such as `selectWatchedToken` with arguments for chain and account?